src/lib/loaders/batchLoader.ts

```diff
@@ -69,26 +111,40 @@ export const batchLoader = ({
   if (!ENABLE_RESOLVER_BATCHING) {
     return singleLoader ? singleLoader : multipleLoader
   }
-  const dl = new DataLoader(keys => {
-    let groupedKeys = groupKeys(keys as any)
-
-    return Promise.all(
-      groupedKeys.map(keys => {
-        if (keys.id.length === 1 && singleLoader) {
-          return singleLoader(keys.id[0])
-        } else {
-          return multipleLoader(keys)
-        }
-      })
-    ).then(data => {
-      const normalizedResults = data.map((queriedGroup, groupIndex) => {
-        return groupedKeys[groupIndex].id.map(
-          id => queriedGroup.find(r => r._id === id) || defaultResult
+  const dl = new DataLoader(
+    (keys: Key[]) => {
+      const normalKeys = normalizeKeys(keys)
+      const [keyGroups, groupedKeys] = groupKeys(normalKeys)
+
+      return Promise.all(
+        groupedKeys.map(compactKeyForRequest).map(keys => {
+          if (keys.id.length === 1 && singleLoader) {
+            return singleLoader(keys.id[0])
+          } else {
+            return multipleLoader(keys)
+          }
+        })
+      ).then((data: Array<GravityResult | GravityResult[]>) => {
+        const normalizedData = data.map(
+          datum => (Array.isArray(datum) ? datum : [datum])
         )
+
+        return normalKeys.map(key => {
+          const group = getKeyGroup(key)
+          const groupIndex = keyGroups.indexOf(group)
+          return (
+            normalizedData[groupIndex].find(
+              ({ _id, _slug }) => key.id === _id || key.id === _slug
+            ) || defaultResult
+          )
+        })
       })
-      return flatten(normalizedResults)
-    })
-  })
+    },
+    {
+      maxBatchSize: 20,
```

Comment:
I was previously naively splitting batch sizes inside of the dataloader. This eliminates the need for doing that (making the overall logic a lot simpler)